hugo/content/docs/langium-overview.md

```diff
@@ -0,0 +1,87 @@
+
+```ts
+Person: 
+    'person' // keyword 
+    name=ID // semantic assignment
+; 
+```
+
+Parses a simple object `{ name: 'John' }`. To interact with this object in a type safe manner, the `langium-cli` tool generates TypeScript type definitions from it. The `Person` parser rules generates the following interface:
+
+```ts
+interface Person {
+    name: String
+}
+```
+
+These interfaces allow you to traverse your abstract syntax tree safely. In case your grammar changes, they will also notify you of any breaking changes which might require you to change your domain logic.
+
+## Cross References and Linking
+
+You can reference another element of your DSL using the grammar cross reference feature:
+
+```ts
+Hello: 'hello' person=[Person];
+```
+
+This `Hello` rule references another `Person` element. By default, the `name` property of an object is used as its identity. Consequently, a `person John` object will be referenced using the `hello John` snippet. Any other name will result in a linking error.
+
+Cross references and linking is accomplished using the concept of 'scoping'. You likely know scopes from programming, where certain variables are only available from certain scopes:
+
+```ts
+let x = 42;
+x = 3; // References the `x` defined in the previous line
+```
+
+```ts
+if (something) {
+    let y = 42;
+}
+y = 3; // Cannot link, `y` isn't in a visible scope
+```
+
+The same occurs in Langium. To enable more complex scoping behavior, you can create a custom domain scoping. For example, common object oriented.languages need a more involved scoping mechanism to resolve references to fields and methods of a class:
+
+```ts
+class X {
+    y(): void { ... }
+}
+
+const instance = new X(); // Symbol `X` is in the local scope
+instance.y(); // Symbol `y` exists in the scope of the `X` class
+```
+
+Aside from those domain specific scoping rules, parsing and resolving cross references is handled completely automatically by Langium.
+
+## Workspace Management
+
+Like with common modularized programming languages, domain logic written in your DSL will usually be split across multiple files to facilitate ease of use and maintenance efforts. This is also possible using Langium, which performs workspace management for you.
+
+When running a Langium based language in a [language server](https://microsoft.github.io/language-server-protocol/), all files in your workspace matching your DSL id will automatically be picked up and processed. In addition, any changes in your workspace will be handled as well. Dealing with added, changed or deleted files in a workspace with multiple hundreds of files can become complicated and decrease performance drastically if not done correctly. Langium employs heuristics to only invalidate and recompute what is actually necessary.
+
+The workspace management also keeps track of the global scope. This allows users of your DSL to reference elements across files within the same workspace.
```

Comment:
Would be great to expand on this. You mentioned scope in the prior section, is the same scope or a different one? Also, would be excellent to add an example case along with a larger explanation; one that motivates the reason why this is important/helpful.